crates/ethereum/evm/src/eip6110.rs

```diff
@@ -0,0 +1,75 @@
+// EIP-6110 parsing
+use alloy_consensus::Request;
+use alloy_eips::eip6110::{DepositRequest, MAINNET_DEPOSIT_CONTRACT_ADDRESS};
+use alloy_sol_types::{sol, SolEvent};
+use reth_interfaces::executor::BlockValidationError;
+use reth_primitives::Receipt;
+use revm_primitives::Log;
+
+pub(crate) fn parse_deposits_from_receipts(
+    receipts: &[Receipt],
+) -> Result<Vec<Request>, BlockValidationError> {
+    let res = receipts
+        .iter()
+        .flat_map(|receipt| receipt.logs.iter())
+        // No need to filter for topic because there's only one event and that's the Deposit event
+        // in the deposit contract.
+        .filter(|log| log.address == MAINNET_DEPOSIT_CONTRACT_ADDRESS)
+        .map(|log| DepositEvent::decode_log(log, false))
+        .map(|res| {
+            let deposit = parse_deposit_from_log(&res?);
+            Ok(Request::DepositRequest(deposit))
+        })
+        .collect::<Result<Vec<_>, _>>()
+        // ugly
+        .map_err(|err: alloy_sol_types::Error| {
+            BlockValidationError::DepositRequestDecode(err.to_string())
+        })?;
+    Ok(res)
+}
+
+sol! {
+    #[allow(missing_docs)]
+    event DepositEvent(
+        bytes pubkey,
+        bytes withdrawal_credentials,
+        bytes amount,
+        bytes signature,
+        bytes index
+    );
+}
+
+fn parse_deposit_from_log(log: &Log<DepositEvent>) -> DepositRequest {
+    // SAFETY: These `expect` https://github.com/ethereum/consensus-specs/blob/5f48840f4d768bf0e0a8156a3ed06ec333589007/solidity_deposit_contract/deposit_contract.sol#L107-L110
```

Comment:
great

Reply:
nit: should be NOTE or something else, SAFETY should be reserved for unsafe {}

Reply:
# Panics in the doc of the function actually